### ISSUE TITLE
customization example doesn't seem to have an effect

### ISSUE THREAD

**Issue by jonathanlocke** [JFormDesigner/FlatLaf#257]
the colors of JTabbedPane can be customized with this:

        UIManager.put("TabbedPane.minimumTabWidth", 150);
        UIManager.put("TabbedPane.background", DARK_GRAY);
        UIManager.put("TabbedPane.foreground", DARK_YELLOW);
        **UIManager.put("TabbedPane.selectedBackground", CHERRY);**
        UIManager.put("TabbedPane.hoverColor", DARK_GRAY.);
        UIManager.put("TabbedPane.underlineColor", DARK_YELLOW);

and most of it works, but the line in bold (which is on the example page for how to customize flatlaf) doesn't seem to have any effect. selectedForeground doesn't seem to work either. i'm using flatlaf 0.45. thanks!

    Jon

**Comment by DevCharly** [comment]
Hmm, just tried. `TabbedPane.selectedBackground` works for me.

Is it possible that the tabbed pane was focused in your test?
Then `TabbedPane.focusColor` is used for the selected (focused) tab background.

~~~java
UIManager.put("TabbedPane.focusColor", Color.red);
~~~

`TabbedPane.selectedForeground` does indeed not work in your case
because `TabbedPane.foreground` is also set.
This is a bug.

To workaround (until fixed in FlatLaf), use a `ColorUIResource` for `TabbedPane.foreground`:
~~~java
UIManager.put("TabbedPane.foreground", new ColorUIResource(WHITE));
~~~

**Comment by DevCharly** [comment]
`TabbedPane.selectedForeground` is now fixed in main branch.

**Comment by jonathanlocke** [comment]
Thanks very much DevCharly!

**Comment by jonathanlocke** [comment]
Is it possible that there's a problem for the background / selectedBackground as well? I see that one too now. I tried your ColorUIResource workaround for that pair of colors and that didn't seem to do the trick.

**Comment by ju-gem** [comment]
Hi,
I also have problems with `TabbedPane.selectedBackground`. 
`UIManager.put("TabbedPane.selectedBackground", new Color(242, 224, 232));`
doesn't have any effect. When not focused, the color shows, ut as soon as I focus my application window again, the background is default-blue again. May be a bug?
Best regards,
ju-gem

**Comment by DevCharly** [comment]
> When not focused, the color shows, but as soon as the focus is set on my application window, the background is default-blue again. May be a bug?

@ju-gem no, that's exactly how it should work.

- `TabbedPane.focusColor` is used as background for selected tab if tabbedpane component itself has focus
- `TabbedPane.selectedBackground` is used if any other component has focus

If you don't want that your tabbedpane can be focused, then use:
~~~java
tabbedPane.setFocusable( false );
~~~

**Comment by ju-gem** [comment]
Alright, that makes sense. Thanks @DevCharly!

**Comment by DevCharly** [comment]
@jonathanlocke the problem with `TabbedPane.selectedForeground` is fixed in FlatLaf 1.0.

Can't find a problem with `TabbedPane.selectedBackground`. See also my previous post.
